app/src/main/java/com/vci/vectorcamapp/landing/presentation/LandingViewModel.kt

```diff
@@ -23,11 +24,13 @@ class LandingViewModel @Inject constructor(
     private val deviceCache: DeviceCache,
     private val currentSessionCache: CurrentSessionCache,
     private val programRepository: ProgramRepository,
+    private val sessionRepository: SessionRepository,
 ) : CoreViewModel() {
 
     private val _state = MutableStateFlow(LandingState())
     val state: StateFlow<LandingState> = _state.onStart {
         loadLandingDetails()
+        observeIncompleteSessionsCount()
     }.stateIn(
```

Comment:
Do all the loading within one function, because thes functions all run asynchronously so this will introduce risk of race conditions.